src/status_im2/contexts/add_new_contact/events_test.cljs

```diff
@@ -10,47 +13,160 @@
-                :state      :error
-                :error      :uncompressed-key}}})))
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; unit tests (validate-contact fn)
+
+(deftest validate-contact-test
+  (are [i e] (= (core/validate-contact (core/init-contact
+                                        {:user-public-key user-ukey :input i}))
+                (core/init-contact e))
+
+   ""            {:user-public-key user-ukey
+                  :input           ""
+                  :type            :empty
+                  :state           :empty}
+
+   " "           {:user-public-key user-ukey
+                  :input           " "
+                  :type            :empty
+                  :state           :empty}
+
+   ukey          {:user-public-key user-ukey
+                  :input           ukey
+                  :id              ukey
+                  :type            :public-key
+                  :public-key      ukey
+                  :state           :invalid
+                  :msg             :t/not-a-chatkey}
+
+   ens           {:user-public-key user-ukey
+                  :input           ens
+                  :id              ens
+                  :type            :ens
+                  :ens             ens-stateofus-eth
+                  :state           :resolve-ens}
+
+   (str " " ens) {:user-public-key user-ukey
+                  :input           (str " " ens)
+                  :id              ens
+                  :type            :ens
+                  :ens             ens-stateofus-eth
+                  :state           :resolve-ens}
+
+   ckey          {:user-public-key user-ukey
+                  :input           ckey
+                  :id              ckey
+                  :type            :compressed-key
+                  :state           :decompress-key}
+
+   link-ckey     {:user-public-key user-ukey
+                  :input           link-ckey
+                  :id              ckey
+                  :type            :compressed-key
+                  :state           :decompress-key}
+
+   link-ens      {:user-public-key user-ukey
+                  :input           link-ens
+                  :id              ens
+                  :type            :ens
+                  :ens             ens-stateofus-eth
+                  :state           :resolve-ens}))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; event handler tests (set-new-identity, set-new-identity-success/error)
+
+(def db
+  {:multiaccount             {:public-key user-ukey}
+   :networks/current-network "mainnet_rpc"
+   :networks/networks        {"mainnet_rpc"
+                              {:id     "mainnet_rpc"
+                               :config {:NetworkId 1}}}})
+
+(deftest empty-string-test ; no callbacks involved
+  (is (= (core/set-new-identity {:db db} "") {:db db})))
+
+(deftest uncompressed-key-test ; no callbacks involved
```

Comment:
I think it's a tad more idiomatic to use the `testing` macro with the explanation instead of a margin comment. When the test fails and the `testing` macro is used, the error will include the test description, whereas the comment won't be displayed.

```clojure
(deftest empty-string-test
  (testing "no callbacks involved"
    (is (= (core/set-new-identity {:db db} "") {:db db}))))
```

Reply:
This isn't really related to the test names, in retrospect I think this is another section of event handler tests - the ones that don't have callbacks to test (or the tests could be refactored so that the callbacks are entirely separate from the event handler tests).

The easiest thing is to just remove the `; no callbacks involved` comments if you don't mind that.

Reply:
> The easiest thing is to just remove the ; no callbacks involved comments if you don't mind that.

Up to you really :) I think the test is easy enough to understand without a `testing` description.

Reply:
Done, I just removed the comments with a 3-semicolon comment to differentiate the event-handler tests without callbacks.